### ISSUE TITLE
`_clap_count_exprs` must be imported when using macro imports instead of 

### ISSUE THREAD

**Issue by ErichDonGubler** [TeXitoi/structopt#157]
This is an upstream `clap` issue I've resolved with [this PR](https://github.com/clap-rs/clap/pull/1397). As far as I can tell, there's no action that can be taken...yet (maybe mid-January 2019). Once `clap` fixes this issue, `structopt` will need to be updated.

---

If one tries to compile code like the following:

```rust
    use {
        // clap_verbosity_flag::Verbosity,
        structopt::{
            clap::arg_enum,
            StructOpt,
        },
    };

    #[derive(Debug, StructOpt)]
    #[structopt(rename_all = "snake_case")]
    pub(crate) struct Args {
        // #[structopt(flatten)]
        // verbosity: Verbosity,
        #[structopt(raw(possible_values = "&ServiceName::variants()", case_insensitive = "true"))]
        service: ServiceName,
        #[structopt(subcommand)]
        subcommand: Subcommand,
    }

    arg_enum! {
        #[derive(Debug)]
        enum ServiceName {
            Github,
            // Gitlab, // TODO: Look into this!
            // BitBucket, // TODO: Look into this!
        }
    }

    #[derive(Debug, StructOpt)]
    #[structopt(rename_all = "snake_case")]
    pub(crate) enum Subcommand {
        Fork,
        New,
        Delete,
        // Clone, // FIXME: I definitely am interested in exploring this
        // Open, // FIXME: I definitely am interested in exploring this.
    }

```

...then one gets an error like this:

```
error: cannot find macro `_clap_count_exprs!` in this scope
  --> src\main.rs:23:5
   |
23 | /     arg_enum! {
24 | |         #[derive(Debug)]
25 | |         enum ServiceName {
26 | |             Github,
...  |
29 | |         }
30 | |     }
   | |_____^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

The current workaround is to do import `_clap_count_exprs` like so:

```diff
--- src/main.rs.old     2018-12-18 16:43:57.468698400 -0700
+++ src/main.rs 2018-12-18 16:44:05.863379100 -0700
@@ -4,7 +4,10 @@
     use {
         // clap_verbosity_flag::Verbosity,
         structopt::{
-            clap::arg_enum,
+            clap::{
+                _clap_count_exprs,
+                arg_enum,
+            },
             StructOpt,
         },
     };
```

**Comment by TeXitoi** [comment]
I don't get it, what structopt would have to do one time this is fixed in clap? a cargo update will not be enough?

**Comment by hcpl** [comment]
I think this is a heads up that minimal Rust will be bumped with that change, though I proposed another solution that doesn't require that in https://github.com/clap-rs/clap/pull/1397#issuecomment-448519540.

**Comment by TeXitoi** [comment]
The policy of structopt on rustc version is to be compatible with the oldest rustc version compatible with de dependencies. I can't manage the other dependencies, thus if clap update its required version, I have nothing to do as I can do nothing ;-)

**Comment by ErichDonGubler** [comment]
@TeXitoi: The bottom line is that in order to consider the issue "solved" you'd want to bump the `clap` dependency, but that would be it. You would definitely want to note that in any form of release notes, though.

It's certainly possible that this would be picked up "for free" in the absence of a `Cargo.lock` file since Cargo tries to use the latest Semver-compatible dependencies, but you can't guarantee that the papercut the upstream PR fixes will be gone unless the bump is done.

**Comment by TeXitoi** [comment]
Minimum clap is now 2.33 (to be published in v0.3)
